orttraining/orttraining/training_ops/cpu/nn/layer_norm.cc

```diff
@@ -54,9 +55,9 @@ Status LayerNormGrad<T>::Compute(OpKernelContext* op_kernel_context) const {
 
   // Note: Eigen has column-major storage order by default
   ConstEigenArrayMap<T> Y_grad_arr{Y_grad->Data<T>(), M, N};
-  ConstEigenArrayMap<T> X_arr{X->Data<T>(), M, N};
+  ConstEigenArrayMap<T> Y_arr{Y->Data<T>(), M, N};
   ConstEigenVectorArrayMap<T> scale_vec{scale->Data<T>(), M};
-  ConstEigenVectorArrayMap<float> mean_vec{mean->Data<float>(), N};
+  ConstEigenVectorArrayMap<T> bias_vec{bias->Data<T>(), M};
   ConstEigenVectorArrayMap<float> inv_std_var_vec{inv_std_var->Data<float>(), N};
```

Comment:
>inv_std_var [](start = 50, length = 11)

Is this a training only data/kernel? Is this available during inference?

Reply:
There is a LayerNormlization kernel:
https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/contrib_ops/cpu/layer_norm.cc
https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/contrib_ops/cuda/layer_norm.cc

That is also used in GPT-2 inference. Please take some time to avoid gradient computing for inference (some outputs are optional, so do not compute them when they do not exist in node output).

Reply:
This implementation is only for training now. Inference has its own implementation.

Reply:
@tianleiwu this PR is to optimize LayernormGrad. Regarding to Layernorm, we can create another work item for it.

Reply:
@yuslepukhin this is training only kernel.